### PR TITLE
test case for overflows in collectives due to large buffers

### PR DESCRIPTION
**Pull request by KavithaTipturMadhu**
Tests for collectives resulting in overflows for large buffers: Issue #2257

### CODE REVIEW

**Comment by pavanbalaji** [comment]
This needs to be run through the code cleanup script.  Also, it'd be better to make the test more comprehensive (add message sizes starting from 1 byte, add all datatypes, all communicators, etc.)

**Comment by KavithaTipturMadhu** [comment]
Ran code cleanup script,  added tests for all data types and buffer sizes. Communicator is probably an overkill because the number of processes spawned is just 1 or 2 and the issue and the fix have no obvious impact on communicators.

**Comment by raffenet** [comment]
Can you squash these into a single commit? It'll be easier to review that way.

**Comment by KavithaTipturMadhu** [comment]
@raffenet I have squashed them.

File: test/mpi/coll/largebuf.c [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+                        errs++;
+                    }
+                }
+            }
+            free(sendbuf);
+            free(recvbuf);
+            free(sol);
+        }
+    }
+    printf("i shoudlve come here\n");
```
Comment (raffenet) [inline test/mpi/coll/largebuf.c]: need to delete this

**Comment by KavithaTipturMadhu** [comment]
Ran code cleanup script, added tests for all data types and buffer sizes. Communicator is probably an overkill because the number of processes spawned is just 1 or 2 and the issue and the fix have no obvious impact on communicators.

**Comment by pavanbalaji** [comment]
Why is there no obvious impact on communicators?  It's collective and context ID bits impact collectives in many cases.

**Comment by KavithaTipturMadhu** [comment]
Context id datatype has not been modified. The problem is due to integer variables being used to hold sizes of buffers.

**Comment by pavanbalaji** [comment]
Not modified does not mean will not interact.

**Comment by KavithaTipturMadhu** [comment]
There are no interactions, local variables holding buffer sizes are overflowing.

**Comment by pavanbalaji** [comment]
Are all communicators going through the exact same code path?

**Comment by KavithaTipturMadhu** [comment]
Yes, MPI_Comm* is passed to the functions that need fixing.

**Comment by pavanbalaji** [comment]
That was not my question.  Do all communicators (e.g., intra and inter communicators) take the same code path in the collective code (e.g., no additional variables used)?

**Comment by KavithaTipturMadhu** [comment]
Yes

**Comment by pavanbalaji** [comment]
I don't think so.  Please read through the code before answering my questions.  (and feel free to move this discussion to slack)

**Comment by akhillanger** [comment]
You may also not want to run the test for each power of 2 size. This will increase the mpich test time quite a bit. We can run it for selected large sizes.

**Comment by pavanbalaji** [comment]
Should this be a separate patch in #2735 instead of a new PR?

**Comment by KavithaTipturMadhu** [comment]
Closing this one and merging it with the fix patch.

**Pull request closed without merging**